src/WPF/ArcGISRuntime.WPF.Samples/Samples/Data/EditAndSyncFeatures/EditAndSyncFeatures.xaml.cs

```diff
@@ -0,0 +1,511 @@
+            Editing, // A feature is in the process of being moved
+            Ready // The geodatabase is ready for synchronization or further edits
+        }
+
+        // URI for a feature service that supports geodatabase generation
+        private Uri _featureServiceUri = new Uri("https://sampleserver6.arcgisonline.com/arcgis/rest/services/Sync/WildfireSync/FeatureServer");
+
+        // Path to the geodatabase file on disk
+        private string _gdbPath;
+
+        // Task to be used for generating the geodatabase
+        private GeodatabaseSyncTask _gdbSyncTask;
+
+        // Flag to indicate which stage of the edit process we're in
+        private EditState _readyForEdits = EditState.NotReady;
+
+        // Hold a reference to the generated geodatabase
+        private Geodatabase _resultGdb;
+
+        public EditAndSyncFeatures()
+        {
+            InitializeComponent();
+
+            // Create the UI, setup the control references and execute initialization
+            Initialize();
+        }
+
+        private async void Initialize()
+        {
+            // Create a tile cache and load it with the SanFrancisco streets tpk
+            TileCache _tileCache = new TileCache(await GetTpkPath());
+
+            // Create the corresponding layer based on the tile cache
+            ArcGISTiledLayer _tileLayer = new ArcGISTiledLayer(_tileCache);
+
+            // Create the basemap based on the tile cache
+            Basemap _sfBasemap = new Basemap(_tileLayer);
+
+            // Create the map with the tile-based basemap
+            Map myMap = new Map(_sfBasemap);
+
+            // Assign the map to the MapView
+            MyMapView.Map = myMap;
+
+            // Create a new symbol for the extent graphic
+            SimpleLineSymbol lineSymbol = new SimpleLineSymbol(SimpleLineSymbolStyle.Solid, Colors.Red, 2);
+
+            // Create graphics overlay for the extent graphic and apply a renderer
+            GraphicsOverlay extentOverlay = new GraphicsOverlay();
+            extentOverlay.Renderer = new SimpleRenderer(lineSymbol);
+
+            // Add graphics overlay to the map view
+            MyMapView.GraphicsOverlays.Add(extentOverlay);
+
+            // Set up an event handler for when the viewpoint (extent) changes
+            MyMapView.ViewpointChanged += MapViewExtentChanged;
+
+            // Update the local data path for the geodatabase file
+            _gdbPath = GetGdbPath();
+
+            // Create a task for generating a geodatabase (GeodatabaseSyncTask)
+            _gdbSyncTask = await GeodatabaseSyncTask.CreateAsync(_featureServiceUri);
+
+            // Add all graphics from the service to the map
+            foreach (var layer in _gdbSyncTask.ServiceInfo.LayerInfos)
+            {
+                // Create the ServiceFeatureTable for this particular layer
+                ServiceFeatureTable onlineTable = new ServiceFeatureTable(new Uri(_featureServiceUri + "/" + layer.Id));
```

Comment:
This is like a coding "run-on sentence". Break the logic into 2 lines of code: on for the Uri and another for the ServiceFeatureTable.